ttt_board.rb

```diff
@@ -0,0 +1,199 @@
+# class to represent the tic-tac-toe board
+
+
+class Board
+    def initialize
+        # intiailize an empty board, indexed as [column][row]
+        # (note that col:row indexing makes the board harder to display than row:col,
+        # becuase we have to print every third value from each  nested arrays)
+        @the_board = [[nil, nil, nil], [nil, nil, nil], [nil, nil, nil]]
+        #              A1   A2   A3     B1   B2   B3     C1   C2   C3
+    end
+
+
+    def count_valid_moves
+        # count empty slots (remaining moves in the game)
+        counter = 0
+        @the_board.each do |the_col|
+            the_col.each do |the_row|
+                if the_row == nil
+                    counter += 1
+                end
+            end
+        end
+        return counter
+    end
+
+
+    def has_valid_moves?
+        # returns true if players can still make valid moves (board is not completely filled)
+        return count_valid_moves > 0
+    end
+
+
+    def get_position(col, row)
+        # for row and col index, return a readable board position such as 'A1'
+        return index_to_col(col) + index_to_row(row)
+    end
+
+
+    def index_to_col(a_index)
+        # 0 => 'A'
+        # map column index to column as entered by player
+        return (a_index + 'A'.ord).chr
+    end
+
+
+    def index_to_row(a_index)
+        # 0 => '1'
+        # map row index to row as entered by player
+        return (a_index+1).to_s
+    end
+
+
+    def col_to_index(a_col)
+        # 'A' => 0;  'B' => 1;  'C' => 2
+        # input must be validated by caller; assuming it is 'A', 'B' or 'C'
+        return a_col.upcase.ord - 'A'.ord
+    end
+

```

Comment:
Please be consistent with empty lines between methods. IMHO 1 line is enough, especially for such big files.

Reply:
This is my subconscious carry-over from Python, where 2 blank lines between methods are the standard, but that's specific to Python syntax, perhaps unnecessary in Ruby.